[api-minor] Don't print hidden annotations (bug 1815196)

calixteman opened the pull request:
and handle correctly the NoView and NoPrint flags when they're changed
from JS.

Snuffleupagus reviewed (dismissed):
I wonder why we didn't do that originally, since this is consistent with the `mustBeViewed` method just above!?

r=me, assuming all tests pass; thank you!

calixteman commented:
/botio test

calixteman committed:
Don't print hidden annotatons (bug 1815196)

and handle correctly the NoView and NoPrint flags when they're changed
from JS.

Snuffleupagus commented on src/core/annotation.js:
```diff
@@ -628,8 +628,11 @@ class Annotation {
    * @private
    */
   _isPrintable(flags) {
+    // In Acrobat, hidden flag cancels the print one
+    // (see annotation_hidden_print.pdf).
```
*Just to be sure:* Have you checked that this indeed applies to *all* annotation-types, and not just the Widget ones?

calixteman replied:
Yes and I just added a test to make sure we don't have any regressions in the future.

Snuffleupagus commented on src/core/annotation.js:
```diff
@@ -1739,6 +1744,19 @@ class WidgetAnnotation extends Annotation {
+  /** @inheritdoc */
+  _isViewable(flags) {
+    // We don't take into account the `NOVIEW` or `HIDDEN` flags here,
+    // since the visibility can be changed by js coed, hence in case
```
Typo:
```suggestion
    // since the visibility can be changed by js code, hence in case
```

Snuffleupagus commented on src/display/annotation_layer.js:
```diff
@@ -345,21 +345,21 @@ class AnnotationElement {
-          hidden,
-          print: event.detail.display === 0 || event.detail.display === 3,
+          noView: hidden,
+          noPrint: event.detail.display === 1 || event.detail.display === 2,
```
Obviously a pre-existing issue, but can you please add a comment explaining the "magical" values that we compare against in this function?

Snuffleupagus commented on src/display/annotation_layer.js:
```diff
@@ -345,21 +345,21 @@ class AnnotationElement {
         const hidden = event.detail.display % 2 === 1;
```
Nit: We could avoid some repetition in this function by doing the following.
```suggestion
        const { hidden } = event.detail;
        const hidden = display % 2 === 1;
```

Snuffleupagus reviewed (commented):
Before I do a final review, please run all tests to ensure that there's no regressions.

calixteman commented:
/botio test

calixteman commented:
The test `bug1737260-oc` was failing because the visibility of a widget is now handled in the annotation layer, hence I fixed it in adding `annotations: true`.

Snuffleupagus commented:
> The test `bug1737260-oc` was failing because the visibility of a widget is now handled in the annotation layer, hence I fixed it in adding `annotations: true`.

Does that mean that if you use the API directly (and not the full viewer), will rendering now be "wrong" for the document?
I suppose that I don't fully understand exactly why this broke and why updating the *test* is necessary/correct here.

calixteman commented:
/botio test

Snuffleupagus reviewed (approved):
Does this, together with your recent patches, replace PR #15032?

r=me, thank you!

calixteman commented:
> Does this, together with your recent patches, replace PR #15032?
> 
> r=me, thank you!

Yep that's the idea, I just need to rewrite the part to update appearances when background or border colors changed.
